texastribune/donations
# Make BusMem query params work like Circle ones

#### What's this PR do?
+ Makes `?level=<level-name>` work on the BusMem page, just like it does on Circle
+ Consolidates the two `FormBuckets.vue` components into one

#### Why are we doing this? How does it help us?
I'm trying to get this app's front end as clean as possible before it gets a major workout during UMP development. That means consolidating duplicate components and standardizing as much as possible. 

This will also allow our revenue people to send potential customers the BusMem page with a pre-selected level. I also think it makes sense to just use a single `level=` parameter as opposed to separate ones for installment period, amount, etc., given that subsidiary info can be derived from a particular level constant.

Cheers to Val for getting this page off to a wonderful start!

#### How should this be manually tested?
+ If you're not testing on the Heroku branch, do a `yarn` then a `make restart`
+ Visit `/business?level=<bigTexYearly|bigTexMonthly|loneStarYearly|loneStarMonthly|hatsOffYearly|hatsOffMonthly>` and ensure the proper bucket radio button is selected on page load. Make a few test donations and ensure the info comes thru properly in #stripe.
+ Make a few test donations on `/circle` just for good measure. Ensure the info comes thru properly in #stripe.

### How should this change be communicated to end users?
I'll let April know and then she can decide.

#### Are there any smells or added technical debt to note?
No.

#### What are the relevant tickets?
https://3.basecamp.com/3098728/buckets/736178/todos/1493933269

#### Have you done the following, if applicable:
* [ ] Added automated tests? *(someday)*
* [ ] Checked for security implications? *(we're now only processing a single query param, and only processing it if it contains a valid `level` value, so I feel good about things)*
